### PR TITLE
adds function to check partial strings within submitted text

### PR DESCRIPTION
**Pull request by reynoldsryan**
https://joinhoney.atlassian.net/browse/ENG-657

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,22 +1,20 @@
...
-  "version": "0.1.3",
-  "homepage": "https://github.com/raymondjavaxx/swearjar-node",
+  "version": "1.0.0",
+  "homepage": "https://github.com/honeyscience/swearjar-node",
   "keywords": ["swear", "curse", "unallowed", "profanity", "clean", "filter", "badwords"],
   "author": {
-    "name": "Ramon Torres",
-    "email": "raymondjavaxx@gmail.com",
-    "url": "https://github.com/raymondjavaxx"
+    "name": "The Honey Team"
```
Comment (walshie4) [inline package.json]: can you toss our github org url (https://github.com/honeyscience) in here as the `url`?

File: lib/swearjar.js [hunk 2]
```diff
@@ -4,6 +4,22 @@ var swearjar = {
 
   _badWords: {},
 
+  checkForProfanity(text, start = 0, partialLength = 3) {
```
Comment (walshie4) [inline lib/swearjar.js]: I think rather than create a new function I'd prefer to add an `options` or `partial = false` optional param to `profane`.

Comment (walshie4) [inline lib/swearjar.js]: Also a better approach in my opinion would be to use a regex to check the entire string at once.

Ex:
image

Comment (walshie4) [inline lib/swearjar.js]: Could leverage https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some as well to do the checking

**Comment by walshie4** [comment]
Please update the README appropriately once the approach is solidified. Side note: do we want to offer this functionality on all of the available functions or just `profane` for now?

File: lib/swearjar.js (hunk 2 above)
Comment (reynoldsryan) [inline lib/swearjar.js]: Let's discuss when you have a minute.

**Comment by reynoldsryan** [comment]
Updated and we can hold off on updating the other functions for now as I don't see any use for them in our system.

File: package.json [hunk 3]
```diff
@@ -1,22 +1,21 @@
 {
-  "name": "swearjar",
+  "name": "@honeyscience/honey-swearjar",
   "description": "Profanity detection and filtering library.",
-  "version": "0.1.3",
-  "homepage": "https://github.com/raymondjavaxx/swearjar-node",
+  "version": "1.0.0",
```
Comment (walshie4) [inline package.json]: please revert this, will be handled in the publishing process

**Pull request merged**